pkg/rpcmiddleware/validator/validator.go

```diff
@@ -60,17 +60,45 @@ func convertError(err error) error {
 	return grpc.Errorf(codes.InvalidArgument, err.Error())
 }
 
-var errNoValidator = errors.DefineUnimplemented("no_validator", "validator not defined")
+var (
+	errNoValidator = errors.DefineUnimplemented("no_validator", "validator not defined")
+	errNonZeroPath = errors.DefineInvalidArgument("non_zero_path", "path '{path}' is not zero")
+)
 
 func validateMessage(ctx context.Context, fullMethod string, msg interface{}) error {
 	if v, ok := msg.(interface {
 		GetFieldMask() types.FieldMask
 	}); ok {
 		region := trace.StartRegion(ctx, "validate field mask")
-		if forbiddenPaths := forbiddenPaths(v.GetFieldMask().Paths, allowedFieldMaskPaths[fullMethod]); len(forbiddenPaths) > 0 {
+
+		paths := v.GetFieldMask().Paths
+		allowedPaths := allowedFieldMaskPaths[fullMethod]
+		if forbiddenPaths := forbiddenPaths(paths, allowedPaths); len(forbiddenPaths) > 0 {
 			region.End()
 			return errForbiddenFieldMaskPaths.WithAttributes("forbidden_paths", forbiddenPaths)
 		}
+
+		var isZero func(string) bool
+		for _, p := range paths {
+			_, ok := allowedPaths[p]
+			if ok {
+				continue
+			}
+			if isZero == nil {
+				if v, ok := msg.(interface {
+					FieldIsZero(string) bool
+				}); ok {
+					isZero = v.FieldIsZero
+				} else {
+					// TODO: Use reflection to validate a (sub-)field by path
```

Comment:
Do we want to do this?
If so, I will file an issue and add a reference according to the guidelines, otherwise I will remove this `else`